File: files-io-file-operations/src/main/java/l/files/io/file/operations/Size.java
package l.files.io.file.operations;

import java.io.IOException;

import l.files.io.file.FileInfo;
import l.files.logging.Logger;

public final class Size extends Count {

    private static final Logger logger = Logger.get(Size.class);

    private volatile long size;

    public Size(Iterable<String> paths) {
        super(paths);
    }

    /**
     * Gets the size in bytes counted so far.
     */
    public long getSize() {
        return size;
    }

    @Override
    protected void onCount(String path) {
        super.onCount(path);
        try {
            size += FileInfo.get(path).getSize();
        } catch (IOException e) {
            logger.warn(e);
        }
    }
}

File: files-app/src/main/kotlin/operations/Size.kt
package l.files.operations

import java.io.IOException

import l.files.fs.Path
import l.files.fs.Resource

private class Size(paths: Iterable<Path>) : Count(paths) {

    volatile var size = 0L
        private set

    override fun onCount(resource: Resource) {
        super.onCount(resource)
        try {
            size += resource.readStatus(false).size
        } catch (e: IOException) {
            // Ignore count
        }
    }

}
